### PR TITLE
Implement support for glob in rule_files.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: https://github.com/timescale/o11y-team-applications/issues/206

TODO: Update the changelog after `0.12.0` is out.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Review by antekresic — APPROVED** [review]

**Review by niksajakovljevic — APPROVED** [review]

**Review by arajkumar — COMMENTED** [review]
@Harkishen-Singh Changes looks good to me except the commit title. Here we are only supporting glob expansion, but the title says regex. Can the title amended to have only glob instead of regex?

File: pkg/rules/rules_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/pgclient"
+)
+
+func TestRegexInConfig(t *testing.T) {
+	cfg := DefaultConfig
+	cfg.PrometheusConfigAddress = "./testdata/rules.regex.config.yaml"
```
Comment (arajkumar) [inline pkg/rules/rules_test.go:18]:
```suggestion
	cfg.PrometheusConfigAddress = "./testdata/rules.glob.config.yaml"
```

**Comment by alejandrodnm** [comment]
LGTM just missing an entry in the Changelog

**Comment by Harkishen-Singh** [comment]
Thank you for review guys! Will implement the suggestions and merge.

**Review by alejandrodnm — APPROVED** [review]

**Pull request merged**